i18n/shared/src/main/scala/io/udash/i18n/TranslationKey.scala

```diff
@@ -106,4 +142,47 @@ object TranslationKey {
   private class Key8[T1, T2, T3, T4, T5, T6, T7, T8](override val key: String) extends TranslationKey8[T1, T2, T3, T4, T5, T6, T7, T8]
   private class Key9[T1, T2, T3, T4, T5, T6, T7, T8, T9](override val key: String) extends TranslationKey9[T1, T2, T3, T4, T5, T6, T7, T8, T9]
   private class KeyX(override val key: String) extends TranslationKeyX
+
+  private[i18n] class ReducedTranslationKey(override val key: String, val argv: Any*) extends TranslationKey0 {
+    override def apply()(implicit provider: TranslationProvider, lang: Lang): Future[Translated] =
+      provider.translate(key, argv:_*)
+  }
+
+  private[i18n] class Untranslatable(override val key: String) extends TranslationKey0 {
+    override def apply()(implicit provider: TranslationProvider, lang: Lang): Future[Translated] =
+      Future.successful(Translated(key))
+  }
+}
+
+object TranslationKey0 {
+  implicit val codec: GenCodec[TranslationKey0] = new GenCodec[TranslationKey0] {
```

Comment:
You could've used `GenCodec.create`, although I don't think it will ever be different from this.